pkg/sql/pg_catalog.go

```diff
@@ -1428,27 +1428,35 @@ CREATE TABLE pg_catalog.pg_type (
 `,
 	populate: func(_ context.Context, p *planner, addRow func(...parser.Datum) error) error {
 		h := makeOidHasher()
-		for oid, typ := range parser.OidToType {
+		for o, typ := range parser.OidToType {
 			cat := typCategory(typ)
 			typElem := oidZero
 			builtinPrefix := parser.PGIOBuiltinPrefix(typ)
 			if cat == typCategoryArray {
-				typElem = parser.NewDOid(parser.DInt(parser.UnwrapType(typ).(parser.TArray).Typ.Oid()))
-				if typ != parser.TypeIntVector {
+				if typ == parser.TypeIntVector {
+					// IntVector needs a special case because its a special snowflake
+					// type. It's just like an Int2Array, but it has its own OID. We
+					// can't just wrap our Int2Array type in an OID wrapper, though,
+					// because Int2Array is not an exported, first-class type - it's an
+					// input-only type that translates immediately to int8array. This
+					// would go away if we decided to export Int2Array as a real type.
```

Comment:
The downside to exporting TypeInt2Array is that it introduces a funky inconsistency - it'd be the only type that's exported from the package but not supported as a first class type.

The reason what you suggest won't work is that `checkResultType` unwraps the input type before checking for equality, so you need to actually be adding `TypeInt2Array` to the switch instead of `TypeIntVector`. I played around with this a little and decided to do it with this special case because it seemed the least messy.

I'm 50/50 on whether we should just export `TypeInt2Array` despite it being inconsistent - if you think that's fine then I'll go ahead and agree.